source/patterns/@aws-solutions-constructs/aws-apigatewayv2websocket-sqs/lib/index.ts

```diff
@@ -0,0 +1,178 @@
+   */
+  readonly deployDeadLetterQueue?: boolean;
+  /**
+   * Optional user provided properties for the dead letter queue
+   *
+   * @default - Default props are used
+   */
+  readonly deadLetterQueueProps?: sqs.QueueProps;
+  /**
+   * The number of times a message can be unsuccessfully dequeued before being moved to the dead-letter queue.
+   *
+   * @default - required only if deployDeadLetterQueue = true.
+   */
+  readonly maxReceiveCount?: number;
+  /**
+   * API Gateway Request Template for the default route.
+   *
+   * @default - "Action=ReceiveMessage"
+   */
+  readonly defaultRouteRequestTemplate?: { [contentType: string]: string };
+
+  /**
+   * Whether to create a default route. If set to true, then it will use the value supplied with `defaultRouteRequestTemplate`
+   *
+   * @default - false.
+   */
+  readonly createDefaultRoute?: boolean;
+
+  /**
+   * User provided websocket authorizer. If not provided, the constructs defaults to
+   * {@link https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_apigatewayv2.WebSocketAuthorizer.html WebSocketAuthorizer}
+   */
+  readonly defaultAuthorizer?: apigwv2.IWebSocketAuthorizer;
+}
+
+export class ApiGatewayV2WebSocketToSqs extends Construct {
+  public readonly webSocketApi: apigwv2.WebSocketApi;
+  public readonly webSocketStage: apigwv2.WebSocketStage;
+  public readonly apiGatewayRole: iam.Role;
+  public readonly apiGatewayLogGroup: logs.LogGroup;
+  public readonly sqsQueue: sqs.Queue;
+  public readonly deadLetterQueue?: sqs.DeadLetterQueue;
+
+  constructor(scope: Construct, id: string, props: ApiGatewayV2WebSocketToSqsProps) {
+    super(scope, id);
+
+    if (props.existingWebSocketApi && props.webSocketApiProps) {
+      throw new Error("Provide either an existing WebSocketApi instance or WebSocketApiProps, not both");
+    }
+
+    const constructMandatedDlqProps: sqs.QueueProps = {
+      fifo: true,
+      deduplicationScope: sqs.DeduplicationScope.MESSAGE_GROUP,
+      fifoThroughputLimit: sqs.FifoThroughputLimit.PER_MESSAGE_GROUP_ID,
+    };
+
+    const constructMandatedQueueProps: sqs.QueueProps = {
+      fifo: true,
+      deduplicationScope: sqs.DeduplicationScope.MESSAGE_GROUP,
+      fifoThroughputLimit: sqs.FifoThroughputLimit.PER_MESSAGE_GROUP_ID,
+      redriveAllowPolicy: {
+        redrivePermission: sqs.RedrivePermission.DENY_ALL,
+      },
+      visibilityTimeout: cdk.Duration.minutes(15),
+    };
+
+    // Setup the queue
+    const buildQueueResponse = defaults.buildQueue(this, "queue", {
+      existingQueueObj: props.existingQueueObj,
+      queueProps: props.queueProps,
+      deployDeadLetterQueue: props.deployDeadLetterQueue,
+      deadLetterQueueProps: props.deadLetterQueueProps,
+      maxReceiveCount: props.maxReceiveCount,
+      enableEncryptionWithCustomerManagedKey: props.enableEncryptionWithCustomerManagedKey,
+      encryptionKey: props.encryptionKey,
+      encryptionKeyProps: props.encryptionKeyProps,
```

Comment:
any risk of simplifying this to pass `...props` instead for those that are named the same?

Reply:
I don't really care either way, I like the explicitness of this too, its just a lot of props is all...